independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java

```diff
@@ -231,8 +235,32 @@ protected Engine engine() {
         }
 
         private CompletionStage<String> renderAsyncNoTimeout() {
-            StringBuilder builder = new StringBuilder(1028);
-            return renderData(data(), builder::append).thenApply(v -> builder.toString());
+            StringBuilder builder = new StringBuilder(getCapacity());
+            return renderData(data(), builder::append).thenApply(v -> {
+                String str = builder.toString();
+                capacity.update(str.length());
+                return str;
+            });
```

Comment:
If this is on the hotpath, maybe not use a lambda?

Reply:
I seriously doubt that it would help but I can try!

Reply:
Hm, I'm not sure I completely understand. The `renderData()` method returns `CompletionStage<Void>` and when the return value is completed we can be sure no other chunks will be added to the `StringBuilder`. Now, I can check if `renderData()` returns a completed future but still I would need to wrap the `builder.toString()` with `CompletedStage` or similar. So I can save one lambda iff the return value is completed but will need one more allocation anyway...